tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,427 @@
+instantly run the command. If it's false, require the user to press return.
+
+#### `+tab-list`
+
+```hoon
+  ++  tab-list
+    |~  sole-id=@ta
+    *(list (option:auto tank))
+```
+
+Autocomplete options for the command-line session (to match `+command-parser`).
+
+#### `+on-command`
+
+```hoon
+  ++  on-command
+    |~  [sole-id=@ta command=command-type]
+    *(quip card _^|(..on-init))
+```
+
+Called when a valid command is run.
+
+#### `+can-connect`
+
+```hoon
+  ++  can-connect
+    |~  sole-id=@ta
+    *?
+```
+
+Called to determine whether the app may be connected to. For example, you
+may only want the local ship to be able to connect.
+
+#### `+on-connect`
+
+```hoon
+  ++  on-connect
+    |~  sole-id=@ta
+    *(quip card _^|(..on-init))
+```
+
+Called when a connection is to be made.
+
+#### `+on-disconnect`
+
+```hoon
+  ++  on-disconnect
+    |~  sole-id=@ta
+    *(quip card _^|(..on-init))
+```
+
+Called when a previously made connection is to be disconnected.
+
+### `default` core
+
+This core contains the bare minimum implementation of the additional `shoe` arms
+beyond the 10 standard Gall app ams. It is used
+analogously to how the `default-agent` core is used for Gall apps.
+
+### `agent` core
+
+This is a wrapper core designed to wrap the `shoe` core, which has too many
+arms to be a Gall agent core. This turns it into a standard Gall agent core by
+integrating the additional arms into the standard ones.
+
+
+## The `sole` library {#the-sole-library}
+
+`shoe` apps create `shoe-effect` cards of the `[%shoe
+...]` shape, which currently just wrap `sole-effect`s, i.e. instructions for displaying text and producing other effects in the console.
+
+The list of possible `sole-effects` can be found in `/sur/sole.hoon`. A few
+commonly used ones are as follows. `[%txt p=tape]` is used to display a line of text. `[%bel ~]`
+is used to emit a beep. `[%pro sole-prompt]` is used to set the prompt. `[%mor
+p=(list sole-effect)]` is used to emit multiple effects.
+
+For example, a `sole-effect` that beeps and displays `This is some text.` would
+be structured as
+```hoon
+[%mor [%txt "This is some text."] [%bel ~] ~]
+```
+
+## `%shoe` app walkthrough {#shoe-app-walkthrough}
+
+Here we explore the capabilities of the `%shoe` example app and then go through
+the code, explaining what each line does. 
+
+### Playing with `%shoe`
+
+First let's test the functionality of `%shoe` so we know what we're getting
+into.
+
+Start two fake ships, one named `~zod` and the other can have any name - we will
+go with `~nus`. Fake ships run locally are able to see each other, and our
+intention is to connect their `%shoe` apps.
+
+On each fake ship start `%shoe` by entering `|start %shoe` into dojo. You may
+then switch to `%shoe` by pressing `Ctrl-X`, possibly multiple times, which will
+change the prompt to `~zod:shoe>` and `~nus:shoe>`. Enter `demo` from `~zod` and press Enter:
```

Comment:
Actually I think `|start`ing an app automatically dumps you into its command line, if available? So no ctrl-xing needed. (But you might still want to include is for later necessity.)